### PR TITLE
Remove file status inserts

### PR DESCRIPTION
**Pull request by MancunianSam**
We were adding all of the file and consignment status updates for the backend checks results in the API.
This has now been moved to the https://github.com/nationalarchives/tdr-statuses/ lambda function which then calls API update.

I've updated some of the tests and removed others where they're no longer relevant.

I've removed AllowedPuids and Disallowed puids as these are read by the tdr-statuses lambda.

This can't be deployed until the new backend checks are deployed. This will require downtime.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusService.scala [hunk 1]
```diff
@@ -1,13 +1,12 @@
 package uk.gov.nationalarchives.tdr.api.service
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusService.scala:1]: Realise its not your code changes but there are `validStatusTypes` and `validStatusValues` vals in the object, could you add a couple of unit tests to make sure these vals are checked? These are pretty crucial so should at least have some testing, until they get moved to a DB table.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala [hunk 2]
```diff
@@ -171,6 +171,7 @@ object FileMetadataService {
 
   val SHA256ClientSideChecksum = "SHA256ClientSideChecksum"
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala:172]: Realise its not your code, but could you add some unit tests to check all these hardcoded values. They are pretty crucial, and some test would be good to at least to make devs think before changing them.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala [hunk 3]
```diff
@@ -55,10 +48,9 @@ object FileStatusService {
   val ChecksumMatch = "ChecksumMatch"
   val Antivirus = "Antivirus"
   val FFID = "FFID"
+  val Redaction = "Redaction"
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala:51]: Could you update the unit tests to check this value, and update for the other values, as the unit tests on line 236, and 242 look out of date, as missing a few values.

**Review by TomJKing — COMMENTED** [review]
Looks good (provided it works and doesn't break anything! ;) ) Just some very minor comments.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusService.scala (hunk 1 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusService.scala:1]: There's a couple of tests here https://github.com/nationalarchives/tdr-consignment-api/blob/master/src/test/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusServiceSpec.scala#L628 which check the types and values are what we're expecting, is that enough?

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala (hunk 3 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileStatusService.scala:52]: I've updated the unit tests in `FileStatusServiceSpec` to check all of the values.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala (hunk 2 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala:140]: I've added a test in to check this value. The only place it's being used is checking the status of the checksums which is pretty worthless not we're not updating one file at a time but I think I've changed enough for one PR.

File: src/test/scala/uk/gov/nationalarchives/tdr/api/service/FileServiceSpec.scala [hunk 4]
```diff
@@ -173,30 +163,25 @@ class FileServiceSpec extends AnyFlatSpec with MockitoSugar with Matchers with S
 
     when(fileRepositoryMock.getFiles(consignmentId, FileFilters(None)))
       .thenReturn(Future(fileAndMetadataRows))
-    when(fileRepositoryMock.getRedactedFilePairs(consignmentId)).thenReturn(Future(Seq()))
+//    when(fileRepositoryMock.getRedactedFilePairs(consignmentId)).thenReturn(Future(Seq()))
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/service/FileServiceSpec.scala:166]: Can this code be deleted now?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just one last very minor comment.

File: src/test/scala/uk/gov/nationalarchives/tdr/api/service/FileServiceSpec.scala (hunk 4 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/tdr/api/service/FileServiceSpec.scala]: Good spot, yeah, it can, I've removed it.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**